CRM/Core/Payment/PayPalIPN.php

```diff
@@ -223,17 +223,16 @@ public function recur(&$input, &$ids, &$objects, $first) {
         throw new CRM_Core_Exception("Ignore all IPN payments that are not completed");
       }
 
-      $contribution->contact_id = $ids['contact'];
-      $contribution->financial_type_id = $objects['contributionType']->id;
-      $contribution->contribution_page_id = $ids['contributionPage'];
-      $contribution->contribution_recur_id = $ids['contributionRecur'];
-      $contribution->receive_date = $now;
-      $contribution->currency = $objects['contribution']->currency;
-      $contribution->payment_instrument_id = $objects['contribution']->payment_instrument_id;
-      $contribution->amount_level = $objects['contribution']->amount_level;
-      $contribution->campaign_id = $objects['contribution']->campaign_id;
-
-      $objects['contribution'] = &$contribution;
+      // In future moving to create pending & then complete, but this OK for now.
+      // Also consider accepting 'Failed' like other processors.
+      $input['contribution_status_id'] = $contributionStatuses['Completed'];
```

Comment:
Can we use $input['contribution_status_id'] = 'Completed' ?

Reply:
@pradpnayak I think it probably works either way?